Q1. Pearson correlation coefficient is a measure of the linear relationship between two variables. Suppose
you have collected data on the amount of time students spend studying for an exam and their final exam
scores. Calculate the Pearson correlation coefficient between these two variables and interpret the result.

In [1]:
import pandas as pd

In [2]:
df =pd.DataFrame({ 
    'study time(in hours)' :[3,4,5,6,7,8,9,10],
    'exam scores' : [66,70,74,79,83,88,91,96]})

In [3]:
df

study time(in hours)  exam scores
0                     3           66
1                     4           70
2                     5           74
3                     6           79
4                     7           83
5                     8           88
6                     9           91
7                    10           96

In [4]:
df.corr(method='pearson')

study time(in hours)  exam scores
study time(in hours)              1.000000     0.999257
exam scores                       0.999257     1.000000

it suggests a strong linear relationship between the two variables

Q2. Spearman's rank correlation is a measure of the monotonic relationship between two variables.
Suppose you have collected data on the amount of sleep individuals get each night and their overall job
satisfaction level on a scale of 1 to 10. Calculate the Spearman's rank correlation between these two
variables and interpret the result.

In [6]:
import pandas as pd

In [7]:
df = pd.DataFrame({
    'sleep (every night)' :[3,4,5,6,7,8,9],
    'job satisfaction':[2,3,4,5,7,9,10]})

In [8]:
df

sleep (every night)  job satisfaction
0                    3                 2
1                    4                 3
2                    5                 4
3                    6                 5
4                    7                 7
5                    8                 9
6                    9                10

In [9]:
df.corr(method='spearman')

sleep (every night)  job satisfaction
sleep (every night)                  1.0               1.0
job satisfaction                     1.0               1.0

 values closer to +1 indicating a strong positive monotonic relationship

Q3. Suppose you are conducting a study to examine the relationship between the number of hours of
exercise per week and body mass index (BMI) in a sample of adults. You collected data on both variables
for 50 participants. Calculate the Pearson correlation coefficient and the Spearman's rank correlation
between these two variables and compare the results.

In [14]:
import pandas as pd

In [21]:
data = {'BMI': [24.8, 27.5, 26.1, 30.2, 29.3, 25.5, 28.7, 31.5, 33.1, 22.4,
               26.9, 28.1, 31.8, 27.4, 29.6, 25.3, 29.9, 26.8, 30.5, 33.2,
               24.7, 26.3, 28.9, 30.1, 29.2, 26.8, 27.7, 32.4, 31.0, 25.9,
               27.8, 30.5, 29.1, 28.6, 26.2, 28.9, 29.6, 27.1, 31.3, 32.1,
               25.8, 27.3, 30.7, 28.5, 29.8, 26.5, 27.9, 30.0, 28.3, 32.6],
        
        'exercise_time': [4.5, 5.3, 4.2, 7.1, 6.4, 5.5, 6.8, 8.3, 9.5, 3.5,
                          5.9, 6.1, 8.5, 5.2, 7.2, 4.9, 7.5, 5.8, 8.6, 9.6,
                          4.4, 4.7, 6.3, 7.0, 6.3, 5.7, 5.9, 9.1, 8.7, 5.6,
                          6.0, 7.9, 6.2, 6.4, 5.3, 6.9, 7.6, 5.7, 8.2, 9.2,
                          4.6, 5.6, 7.3, 5.4, 7.1, 5.1, 5.8, 7.7, 6.6, 9.8]}



In [22]:
df = pd.DataFrame(data)
print(df.head())


    BMI  exercise_time
0  24.8            4.5
1  27.5            5.3
2  26.1            4.2
3  30.2            7.1
4  29.3            6.4


In [23]:
#spearman correlation coefficient 
df.corr(method = 'spearman')

BMI  exercise_time
BMI            1.000000       0.958043
exercise_time  0.958043       1.000000

In [26]:
#pearson correlation
df.corr(method='pearson')

BMI  exercise_time
BMI            1.000000       0.955063
exercise_time  0.955063       1.000000

in spearman correlation coefficienta  indicates that there is a very strong positive monotonic relationship between exercise time and BMI, meaning that as the number of hours of exercise per week increases, the BMI tends to increase as well. However, it's important to keep in mind that correlation does not necessarily imply causation, and there may be other factors that are influencing this relationship.

in  Pearson correlation coefficient indicates a very strong positive linear relationship between the number of hours of exercise per week and body mass index (BMI) in your sample of 50 participants.

Q4. A researcher is interested in examining the relationship between the number of hours individuals
spend watching television per day and their level of physical activity. The researcher collected data on
both variables from a sample of 50 participants. Calculate the Pearson correlation coefficient between
these two variables.

In [28]:
import pandas as pd
import numpy as np


In [29]:
# Generate random data for number of hours spent watching TV and level of physical activity
hours_tv = np.random.uniform(low=0, high=8, size=50) # assuming people spend between 0 to 8 hours per day watching TV
activity_level = np.random.randint(low=1, high=6, size=50) # assuming a scale of 1-5 for activity level


In [30]:
data = {'Hours of TV': hours_tv, 'Activity Level': activity_level}


In [31]:
df =pd.DataFrame(data)
df.head()

Hours of TV  Activity Level
0     4.001789               4
1     5.721505               2
2     3.093658               3
3     4.475463               3
4     5.316791               4

In [32]:
#pearson correlaation coefficent
df.corr(method='pearson')


Hours of TV  Activity Level
Hours of TV         1.00000         0.08135
Activity Level      0.08135         1.00000

in Pearson correlation coefficient indicates a very strong positive linear relationship between the  number of hours individuals
spend watching television per day and their level of physical activity.

Q5. A survey was conducted to examine the relationship between age and preference for a particular
brand of soft drink. The survey results are shown below:


In [33]:
import pandas as pd

In [42]:
df =pd.DataFrame({
    
    'Age(Years)':[25 ,42,37,19,31,28],
    'Soft drink Preference':['Coke','Pepsi','Mountain dew','Coke','Pepsi','Coke'],
})

In [43]:
df.head()

Age(Years) Soft drink Preference
0          25                  Coke
1          42                 Pepsi
2          37          Mountain dew
3          19                  Coke
4          31                 Pepsi

In [44]:
df.corr(method ='spearman')


/tmp/ipykernel_1565/2969927184.py:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  df.corr(method ='spearman')


Age(Years)
Age(Years)         1.0

In [45]:
df.corr(method = 'pearson')

/tmp/ipykernel_1565/3294333281.py:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  df.corr(method = 'pearson')


Age(Years)
Age(Years)         1.0

Q6. A company is interested in examining the relationship between the number of sales calls made per day
and the number of sales made per week. The company collected data on both variables from a sample of
30 sales representatives. Calculate the Pearson correlation coefficient between these two variables.

In [47]:
import pandas as pd
import numpy as np


In [48]:

# Generate random data for sales calls and number of sales
sales_calls = np.random.randint(low=50, high=200, size=30) # assuming sales reps make between 50-200 calls per day
sales_made = np.random.randint(low=5, high=30, size=30) # assuming sales reps make between 5-30 sales per week



In [49]:
# Create a dictionary to store the data
data = {'Sales Calls': sales_calls, 'Sales Made': sales_made}



In [50]:
df =pd.DataFrame(data)

In [51]:
df.head()

Sales Calls  Sales Made
0          157          12
1          192          23
2          176          15
3           70          29
4          148          15

In [52]:
#pearson correlation
df.corr(method ='pearson')

Sales Calls  Sales Made
Sales Calls     1.000000    0.168451
Sales Made      0.168451    1.000000

in pearson correlation coefficient of 0.168451 suggests a weak positive correlation between two variables. The closer the value of the correlation coefficient is to 0, the weaker the correlation. In this case, a value of 0.168451 indicates that there is a positive relationship between the two variables, but it is not very strong. 